prepare-image.sh

```diff
@@ -20,9 +20,10 @@ fi
 
 # SOURCE install #
 if [[ $INSTALL_TYPE == "source" ]]; then
-    BUILD_DEPS="python3-pip python3-devel gcc git-core"
+    BUILD_DEPS="python3-devel gcc git-core"
     dnf upgrade -y
-    dnf install -y $BUILD_DEPS
+    # NOTE(dtantsur): pip is a requirement of python3 in CentOS
+    dnf install -y python3-pip $BUILD_DEPS
```

Comment:
Not directly related to this PR, but I wonder do we still need the pinned `21.3.1` version below, or can we just use the packaged version?

Reply:
we could, although packaged version is different between distributions and in certain cases not recent enough to deal with some recent builds.
Since we're using source code for most of the ironic software/libraries/dependencies, I'd rather use also a fixed recent pip version.